src/components/multiSelect/MultiSelectBottomSheet.tsx

```diff
@@ -0,0 +1,271 @@
+import { typeScale } from 'src/styles/fonts'
+
+const ITEM_HEIGHT = 60
+const MAX_ITEMS_IN_VIEW = 5
+
+interface MultiSelectBottomSheetProps<T extends string> {
+  forwardedRef: React.RefObject<GorhomBottomSheet>
+  onChange?: BottomSheetProps['onChange']
+  onClose?: () => void
+  onOpen?: () => void
+  handleComponent?: BottomSheetProps['handleComponent']
+  selectedItems: Record<T, boolean>
+  setSelectedItems: (selectedItems: Record<T, boolean>) => void
+  textAndIconMap: Record<T, Omit<ItemProps, 'onPress' | 'isSelected'>>
+  selectAllText: string
+  title: string
+}
+
+function MultiSelectBottomSheet<T extends string>({
+  forwardedRef,
+  onClose,
+  onOpen,
+  selectedItems,
+  setSelectedItems,
+  textAndIconMap,
+  selectAllText,
+  title,
+}: MultiSelectBottomSheetProps<T>) {
+  const { t } = useTranslation()
+
+  // Bottom Sheet Things
+  const scrollViewRef = useRef<ScrollView>(null)
+  const { height } = useSafeAreaFrame()
+  const insets = useSafeAreaInsets()
+  const renderBackdrop = useCallback(
+    (props: BottomSheetDefaultBackdropProps) => (
+      <BottomSheetBackdrop {...props} disappearsOnIndex={-1} appearsOnIndex={0} />
+    ),
+    []
+  )
+  // fires before bottom sheet animation starts
+  const handleAnimate = (fromIndex: number, toIndex: number) => {
+    if (toIndex === -1 || fromIndex === -1) {
+      // ensure that the keyboard dismiss animation starts at the same time as
+      // the bottom sheet
+      Keyboard.dismiss()
+    }
+
+    if (toIndex === 0) {
+      onOpen?.()
+    }
+  }
+  const handleClose = () => {
+    onClose?.()
+  }
+
+  // Multi select logic things
+  const isEveryItemSelected = Object.values(selectedItems).every((isSelected) => isSelected)
+  const allItemIds = Object.keys(selectedItems) as T[]
+  const everyItemSelected = useMemo(
+    () =>
+      allItemIds.reduce(
+        (acc, key) => {
+          acc[key as T] = true
+          return acc
+        },
+        {} as Record<T, boolean>
+      ),
+    [allItemIds]
+  )
+  const noItemSelected = useMemo(
+    () =>
+      allItemIds.reduce(
+        (acc, key) => {
+          acc[key as T] = false
+          return acc
+        },
+        {} as Record<T, boolean>
+      ),
+    [allItemIds]
+  )
+
+  const selectAllItem = renderItem({
+    text: selectAllText,
+    isSelected: isEveryItemSelected,
+    onPress: () => setSelectedItems(everyItemSelected),
+  })
+
+  const items = (Object.entries(selectedItems) as [T, boolean][]).map(([itemId, isSelected]) =>
+    renderItem({
+      text: textAndIconMap[itemId].text,
+      iconUrl: textAndIconMap[itemId].iconUrl,
+      isSelected: isSelected && !isEveryItemSelected,
+      onPress: () => {
+        if (isEveryItemSelected) {
+          setSelectedItems({ ...noItemSelected, [itemId]: isSelected })
+        } else {
+          setSelectedItems({ ...selectedItems, [itemId]: !isSelected })
+        }
+      },
+    })
+  )
+
+  return (
+    <GorhomBottomSheet
```

Comment:
can we reuse BottomSheet or BottomSheetBase components? seems like some of the things are duplicated here, like handleAnimate, renderBackdrop, etc

Reply:
I'll try and see if any sharing can be done.